lib/statsd/instrument/client.rb

```diff
@@ -282,8 +295,11 @@ def capture(&block)
 
   protected
 
-  def datagram_builder
-    @datagram_builder ||= @datagram_builder_class.new(prefix: prefix, default_tags: default_tags)
+  def datagram_builder(no_prefix:)
```

Comment:
Adding the no-prefix option here limits the scope of adding support for this option: the datagram builder class doens't have to be aware of its existence. It's not super super elegant though :/